src/lints/manual/helpers.rs

```diff
@@ -363,3 +364,75 @@ pub fn expr_if_get_var_name_and_err(expr_if: AstExprIf, db: &dyn SyntaxGroup) ->
 
     (option_var_name, err)
 }
+
+/// Checks if the expr is a function call of NEVER type
+pub fn is_never_type(expr: &Expr, db: &dyn SemanticGroup, arenas: &Arenas) -> bool {
+    let function_call = match expr {
+        // If it is block, it is necessary to extract tail from it
+        Expr::Block(expr_block) => {
+            let Some(tail_expr_id) = expr_block.tail else {
+                return false;
+            };
+            let Expr::FunctionCall(function_call) = &arenas.exprs[tail_expr_id] else {
+                return false;
+            };
+            function_call
+        }
+        Expr::FunctionCall(function_call) => function_call,
+        _ => return false,
+    };
+
+    function_call.ty.short_name(db) == NEVER
+}
+
+/// Checks whether the specified arm of a `match` expression contains
+/// **at most one** statement in its block expression
+pub fn is_match_arm_single_or_empty(
```

Comment:
Maybe something like:
```suggestion
pub fn match_with_single_statement_or_empty(
```

because i don't really like the name of this function